docker_tests/test_prod_image.py

```diff
@@ -127,7 +127,7 @@ def test_pip_dependencies_conflict(self):
         "cncf.kubernetes": ["kubernetes", "cryptography"],
         "dask": ["cloudpickle", "distributed"],
         "docker": ["docker"],
-        "elasticsearch": ["elasticsearch", "es.elastic", "elasticsearch_dsl"],
+        "elasticsearch": ["elasticsearch"],
```

Comment:
yeah so we need also to set milestone on this one for Airflow 2.7.0?

Reply:
I guess.. No need because we aren't going to cherry-pick that one I guess. It would have been more complex if we actually decided to cherry-pick because we woud have to remember to cherry-pick both together (that's what you loose when you split such related PRs that they become non-cherry-pickable as single PR.

Reply:
@eladkal -> do you still want to split it ? And @Owen-CH-Leung - do you understand what we are asking for here? 

I have personally a feeling, the need for splitting such PR will be difficult to explain to the contributors, because they have no context and might be a bit difficult to understand what we are asking for and why and how to do it.

I am - honestly - not sure if this is a better solution, taking into account that the only side-effect of having this ina a single PR is that the same change will appear in both provider and the core.

Reply:
@potiuk pretty hard for me to understand as I don't have experience with releasing airflow & airflow provider package. My guess is : it would be better to have one PR for code change (including tests), and another PR for deprecating the dependencies ? So that means with the code-change PR being merged, we still have non-official elasticsearch library as dependency, just that we are not using them ?

More specifically , the changes to `CHANGELOG.rst`, `test_prod_image.py`, `provider_dependencies.json`, `index,rst` and etc...All go to another PR ? 

I hope I'm getting it correctly but if not please explain to me. I can accommodate it.

Reply:
> @eladkal -> do you still want to split it 

I'm OK with keeping it as is.